store/store.go

```diff
@@ -0,0 +1,127 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+)
+
+var (
+	// ErrNotFound is returned if a Store cannot find the Alert.
+	ErrNotFound = errors.New("alert not found")
+)
+
+// Alerts implements Store using an in-memory map.
+type Alerts struct {
+	sync.Mutex
+	c  map[model.Fingerprint]*types.Alert
+	cb func([]*types.Alert)
+}
+
+// NewAlerts returns a new Alerts struct.
+func NewAlerts(ctx context.Context, gcInterval time.Duration) *Alerts {
+	a := &Alerts{
+		c:  make(map[model.Fingerprint]*types.Alert),
+		cb: func(_ []*types.Alert) {},
+	}
+
+	if gcInterval == 0 {
+		gcInterval = time.Minute
```

Comment:
How about making this default more explicit by referencing a variable at the top of the file:

```golang
var defaultGCInterval = time.Minute
```

This could even be exported and then used in `main.go` for `*alertGCInterval`, right?